haystack/components/converters/markdown.py

```diff
@@ -50,23 +50,22 @@ def run(self, sources: List[Union[str, Path, ByteStream]], meta: Optional[List[D
         Reads text from a markdown file and executes optional preprocessing steps.
 
         :param sources: A list of markdown data sources (file paths or binary objects)
-        :param meta: Optional list of metadata to attach to the Documents.
-          The length of the list must match the number of paths. Defaults to `None`.
+        :param meta: Optional metadata to attach to the Documents.
+          This value can be either a list of dictionaries or a single dictionary.
+          If it's a single dictionary, its content is added to the metadata of all produced Documents.
```

Comment:
Type annotation right now still is `meta: Optional[List[Dict[str, Any]]]` but should be meta: Optional[Union[List[Dict[str, Any]],Dict[str, Any]]] instead no? Same in the [other PR](https://github.com/deepset-ai/haystack/pull/6615)

Reply:
Argh good catch! I missed the types in a couple of other converters, I'll fix them all in this PR